src/ros/tf/republisher.cpp

```diff
@@ -203,8 +204,8 @@ void TFRepublisher::set_initial_poses(double unix_time)
 	if(append_opts) {
 		bson_destroy(append_opts);
 	}
-	if(collection) {
-		delete collection;
+	if(collection_) {
+		collection_.reset();
```

Comment:
reset is not needed, just remove it everywhere. smart pointer deletes itself, no reset needed.